### DISCUSSION TITLE
Dynamic Queues

### DISCUSSION THREAD

**Discussion by frank81** [mikopbx/Core#277] [General ( Обо всём )]
Hi everyone, I'm trying to create dynamic queues but Miko doesn't let me create agentless queues.
This was my configuration with dynamic agents on old asterisk:

queues.conf
[drmercuri]
musicclass = default
strategy = ringall
timeout = 30
retry = 5
wrapuptime = 10
maxlen = 0
announce-frequency = 90
min-announce-frequency = 15
announce-holdtime = one
announce-position = more
announce-position-limit = 10
announce-round-seconds = 30
relative-periodic-announce = yes
reportholdtime = yes
ringinuse = no
joinempty = yes
leavewhenempty = yes

extensions.conf
exten => * 50,1, Verbose (2, Logging In Queue Member)
same => n, Set (MemberChannel {CHANNEL (channeltype)} = $ {CHANNEL (peername)})
same => n, AddQueueMember (drmercuri, $ {MemberChannel})
same => n, Verbose (1, $ {AQMSTATUS}); ADDED, MEMBERALREADY, NOSUCHQUEUE
same => n, Playback (agent-loginok)
same => n, Hangup ()
exten => * 51,1, Verbose (2, Logging Out Queue Member)
same => n, Set (MemberChannel {CHANNEL (channeltype)} = $ {CHANNEL (peername)})
same => n, RemoveQueueMember (drmercuri, $ {MemberChannel})
same => n, Verbose (1, $ {RQMSTATUS}); REMOVED, NOTINQUEUE, NOSUCHQUEUE
same => n, Playback (agent-loggedoff)
same => n, Hangup ()

Can someone help me? Thank you

**Comment by boffart** [comment]
Just create a queue with an unregistered agent **200001**.
See [instructions](https://translate.yandex.ru/translate?url=https%3A%2F%2Fwiki.mikopbx.ru%2Ffaq%3Aqueue_dynamic_agents&lang=ru-en) 

Register a softphone under the number **200001**
Cancel the agent's registration in the queue. To do this, call the number 9990Queue_Number

**Comment by frank81** [comment]
Thanks for the reply,
so if there are no agents in the queue can i tell them to play an audio file?
I used to do this: 
[drmercuri]
exten => s,1,NoOp(${CALLERID(num)})
exten => s,n,Answer()
exten => s,n(giorno),Queue(drmercuri)
exten => s,n,Playback(custom/closed)
exten => s,n,Hangup()
exten => s,n(busy),Busy()

Thanks

**Reply by boffart** [reply]
In the queue settings, use the option "**If there are no agents in the queue, redirect the call to**"
Direct the call to the IVR to play the audio file
<img width="1128" alt="MikoPBX_Call_queue" src="https://user-images.githubusercontent.com/5922739/155279763-4ebc650e-712d-44b0-9bbc-867296bebf04.png">

**Comment by frank81** [comment]
Thank you, you have been very kind
